purchase_request/models/purchase_request.py

```diff
@@ -285,11 +321,15 @@ def button_to_approve(self):
         return self.write({"state": "to_approve"})
 
     def button_approved(self):
-        return self.write({"state": "approved"})
+        if not self.is_approvable:
+            raise UserError(_("You are not allowed to approve this request."))
+        return self.sudo().write({"state": "approved"})
 
     def button_rejected(self):
-        self.mapped("line_ids").do_cancel()
-        return self.write({"state": "rejected"})
+        if not self.is_approvable:
+            raise UserError(_("You are not allowed to reject this request."))
+        self.mapped("line_ids").sudo().do_cancel()
```

Comment:
Relational fields don't need mapped
```suggestion
        self.line_ids.sudo().do_cancel()
```